app/controllers/notifications_controller.rb

```diff
@@ -0,0 +1,16 @@
+class NotificationsController < ApplicationController
+  def index
+    @notifications = Notification.where(user: current_user)
+                                 .includes({ notificationable: %i[user post] })
+                                 .order(id: :desc)
+
+    @notifications = @notifications.where(notificationable_type: params[:type]) if params[:type] == 'Comment' ||
+                                                                                   params[:type] == 'Post' ||
+                                                                                   params[:type] == 'Reaction'
+
+    @notifications.unviewed.update(viewed: true)
+    @unviewed_notifications_count = Notification.where(user: current_user).unviewed.count
```

Comment:
why should you count un-viewed notis after update it becomes viewed right above?

Reply:
because if notis has filtered then maybe some notis still not be view, so i have to count un-viewed after update in case some notis are not viewed.